src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -376,8 +376,22 @@ class MacroAssembler: public Assembler {
     return ((predecessor & 0x3) << 2) | (successor & 0x3);
   }
 
+  void fence(uint32_t predecessor, uint32_t successor) {
+    if (UseZtso) {
+      if (pred_succ_to_membar_mask(predecessor, successor) & StoreLoad) {
```

Comment:
This should be "(pred_succ_to_membar_mask(predecessor, successor) & StoreLoad) == 0".

Reply:
Not AFAIU, as we would then _not_ generate `StoreLoad` barriers which are the only one TSO doesn't guarantee.

Reply:
No, sorry I mean != 1. We don't if on integer values.